xhr/resources/infinite-redirects.py

```diff
@@ -1,22 +1,24 @@
+from wptserve.utils import isomorphic_encode
+
 def main(request, response):
-    location = "%s://%s%s" % (request.url_parts.scheme,
-                              request.url_parts.netloc,
-                              request.url_parts.path)
-    page = "alternate"
+    location = u"%s://%s%s" % (request.url_parts.scheme,
+                               request.url_parts.netloc,
+                               request.url_parts.path)
+    page = u"alternate"
     type = 302
     mix = 0
-    if request.GET.first("page", None) == "alternate":
-        page = "default"
+    if request.GET.first(b"page", None) == b"alternate":
+        page = u"default"
 
-    if request.GET.first("type", None) == "301":
+    if request.GET.first(b"type", None) == b"301":
         type = 301
 
-    if request.GET.first("mix", None) == "1":
+    if request.GET.first(b"mix", None) == b"1":
         mix = 1
         type = 302 if type == 301 else 301
 
-    new_location = "%s?page=%s&type=%s&mix=%s" % (location, page, type, mix)
-    headers = [("Cache-Control", "no-cache"),
-               ("Pragma", "no-cache"),
-               ("Location", new_location)]
-    return 301, headers, "Hello guest. You have been redirected to " + new_location
+    new_location = u"%s?page=%s&type=%s&mix=%s" % (location, page, type, mix)
+    headers = [(b"Cache-Control", b"no-cache"),
+               (b"Pragma", b"no-cache"),
+               (b"Location", isomorphic_encode(new_location))]
+    return 301, headers, b"Hello guest. You have been redirected to " + isomorphic_encode(new_location)
```

Comment:
Can we keep in text and avoid the `isomorphic_encode` here? E.g.

```suggestion
    return 301, headers, u"Hello guest. You have been redirected to " + new_location
```

Reply:
Code updated. Thank you!